src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -137,6 +137,7 @@ internal class CompletionHandler : IRequestHandler<CompletionParams, SumType<Com
                 result = SetResolveData(result.Value, serverKind);
                 if (serverKind == LanguageServerKind.CSharp && string.Equals(request.Context.TriggerCharacter, "@", StringComparison.Ordinal))
                 {
+                    result = DoNotPreselect(result.Value);
```

Comment:
This concerns me a bit. We are basically saying never preselect on all cases. Does this result in a poor completion experience? I would like `DateTime` to be auto-selected when I type `DateT` so I can just press enter or tab. I wouldn't want to press the down arrow key every time I want to commit a completion.

Reply:
This only pre-selects in cases when `@` is the trigger. Meaning as you continue to type and get additional completion items for literals it'll work as expected.

Reply:
> We are basically saying never preselect on all cases

`string.Equals(request.Context.TriggerCharacter, "@", StringComparison.Ordinal)`

I think this should only apply for implicit expressions? On that note though, do we need to handle explicit expressions?

Reply:
![image](https://user-images.githubusercontent.com/14852843/91620385-aac05200-e944-11ea-9d5d-bfdb6bcbecae.png)
Beat me by 13 seconds 😄

Reply:
I'm blind. Ignore me 🙈